### PR TITLE
Upgrade Airflow to 2.6

### PR DESCRIPTION
**Pull request by antidipyramid**
## Overview

This PR upgrades Airflow to the latest version (2.6.0).

### Checklist

- [x] PR has a descriptive enough title to be useful in changelogs

### Notes

A couple of things not reflected in the title and/or code:

- I changed the way our deployment configs are organized so that they're less annoying to deal with. I also moved deployment secrets into an encrypted .env file and decrypted the Supervisor settings.
- I upgraded Python on the staging server following these instructions: https://cloudbytes.dev/snippets/upgrade-python-to-latest-version-on-ubuntu-linux
- Contrary to @antidipyramid's surgical approach, I dropped the Airflow database before deploying to sidestep those upgrade issues.

## Testing Instructions

These changes have been deployed to the staging site at https://la-metro-dashboard-staging.datamade.us

 * Run `docker-compose build`
 * Run `docker-compose up` and verify the Airflow version at `localhost:8080`

Handles #95

### CODE REVIEW

File: airflow.cfg [hunk 1]
```diff
@@ -133,7 +44,7 @@ load_examples = False
 plugins_folder = /app/plugins
 
 # Secret key to save connection passwords in the db
-fernet_key = 
+fernet_key = 38H0RL0YR-fNb2YTbNIAD9CX-1NBcnHNwpJzGlLY5X4=
```
Comment (hancush) [inline airflow.cfg:47]: Should this be public?

File: airflow.cfg [hunk 2]
```diff
@@ -278,7 +295,7 @@ worker_refresh_interval = 30
 
 # Secret key used to run your flask app
 # It should be as random as possible
-secret_key = temporary_key
+secret_key = 28d1c8bda62bb47e67d935c936e301a9f5ddaf40a5de875cf6b7765811b3
```
Comment (hancush) [inline airflow.cfg:298]: And here?

File: dags/check_meeting_broadcast.py [hunk 3]
```diff
@@ -30,5 +30,4 @@
     t1 = BlackboxDockerOperator(
         task_id='check_current_meeting',
         command='python manage.py check_current_meeting',
-        tag=LA_METRO_DOCKER_IMAGE_TAG,
```
Comment (hancush) [inline dags/check_meeting_broadcast.py:33]: Why remove this?

File: dags/scrape_factory.py [hunk 4]
```diff
@@ -47,7 +47,7 @@ def get_dag_id(dag_name, dag_config, interval):
         return '{0}_{1}_thru_{2}'.format(dag_name, int_day_map[days[0]], int_day_map[days[-1]])
 
 @provide_session
-def previous_scrape_done(**kwargs):
+def previous_scrape_done(session=None, **kwargs):
     running_scrapes = kwargs['session'].query(TaskInstance).filter(
```
Comment (hancush) [inline dags/scrape_factory.py:51]: Needed to do this, for this to work:

```suggestion
    running_scrapes = session.query(TaskInstance).filter(
```

File: dags/scrape_factory.py [hunk 5]
```diff
@@ -86,13 +86,12 @@ def previous_scrape_done(**kwargs):
             check_previous = ShortCircuitOperator(
                 task_id='check_previous',
                 python_callable=previous_scrape_done,
-                provide_context=True
             )
 
             scrape = BlackboxDockerOperator(
```
Comment (hancush) [inline dags/scrape_factory.py:91]: Docker operators aren't working anymore. We'll have to figure out why before deploying for testing.

**Review by hancush — CHANGES_REQUESTED** [review]
Thanks, @antidipyramid! Left a few comments inline. If you have cycles to investigate Docker failures this week, that would be awesome. Let me know if not, and I can pick up that line of inquiry.

File: airflow.cfg (hunk 2 above)
Comment (antidipyramid) [reply airflow.cfg:298]: Similar to the Fernet key, I've added a [Flask secret key environment variable](https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html?highlight=configuration#config-webserver-secret-key). 

The docs list three environment variables but it seems like you would only have to set one?

**Comment by hancush** [comment]
Followed these instructions to upgrade Python on the server: https://cloudbytes.dev/snippets/upgrade-python-to-latest-version-on-ubuntu-linux

File: .github/workflows/main.yml [hunk 6]
```diff
@@ -28,18 +28,22 @@ jobs:
...
     steps:
     - uses: actions/checkout@v2
-    - name: Set up Python 3.6
+    - name: Set up Python
       uses: actions/setup-python@v2
+      id: setup-python
       with:
-        # Semantic version range syntax or exact version of a Python version
-        python-version: '3.7'
+        python-version: '3.10'
```
Comment (hancush) [inline .github/workflows/main.yml:35]: Upgraded Python in GitHub Actions.

File: .blackbox/blackbox-files.txt [hunk 7]
```diff
@@ -1,2 +1,2 @@
-configs/la-metro-dashboard.production.conf.supervisor
-configs/la-metro-dashboard.staging.conf.supervisor
+configs/production/la-metro-dashboard.conf.supervisor
+configs/staging/.env
```
Comment (hancush) [inline .blackbox/blackbox-files.txt:2]: Switched from encrypted Supervisor config to encrypted secrets.

File: Dockerfile [hunk 8]
```diff
@@ -1,6 +1,5 @@
-FROM python:3.7
+FROM python:3.10
```
Comment (hancush) [inline Dockerfile:1]: Upgraded Python in our Docker image.

File: dags/airflow_db_cleanup.py [hunk 9]
```diff
@@ -1,268 +0,0 @@
-"""
-A maintenance workflow that you can deploy into Airflow to periodically clean
```
Comment (hancush) [inline dags/airflow_db_cleanup.py:2]: Don't need these anymore – the Airflow CLI comes with a handy cleanup script we can run.

File: plugins/operators/blackbox_docker_operator.py [hunk 10]
```diff
@@ -35,7 +35,9 @@ def __init__(self, *args, **kwargs):
 
         self.force_pull = True
         self.auto_remove = True
-        self.volumes = list(set(self.DEFAULT_VOLUMES + self.volumes))
+
+        self.mount_tmp_dir = False
```
Comment (hancush) [inline plugins/operators/blackbox_docker_operator.py:39]: Need to do this to permit other mounts.

File: scripts/after_install.sh [hunk 11]
```diff
@@ -3,7 +3,7 @@ set -euxo pipefail
 
 # Make sure the deployment group specific variables are available to this
 # script.
-source ${BASH_SOURCE%/*}/../configs/$DEPLOYMENT_GROUP_NAME-config.conf
+source ${BASH_SOURCE%/*}/../configs/$DEPLOYMENT_GROUP_NAME/deployment.cfg
```
Comment (hancush) [inline scripts/after_install.sh:6]: Related to config organization change.

**Comment by hancush** [comment]
@xmedr @antidipyramid Can y'all take a look at this PR next week?

**Comment by antidipyramid** [comment]
@hancush I've been able to run DAGs locally as expected using Airflow 2.6. Everything seems to be working!

**Pull request merged**